src/Reveal.Sdk.Dom/Data/DataSourceItems/WebServiceDataSourceItem.cs

```diff
@@ -1,19 +1,148 @@
+            this(title, dataSource)
+        {
+            Url = uri;
+        }
+
+        public WebServiceDataSourceItem(string title, DataSource dataSource) :
+            base(title, new JsonDataSource())
+        {
+            _dataSource = dataSource ?? new DataSource();
+            InitializeResourceItem(title);
+            UpdateResourceItemDataSource(_dataSource);
+        }
+
         [JsonIgnore]
         public string Url
         {
             get => Properties.GetValue<string>("Url");
             set => Properties.SetItem("Url", value);
         }
+
+        [JsonIgnore]
+        public bool IsAnonymous
+        {
+            get { return ResourceItemDataSource.Properties.GetValue<bool>("_rpUseAnonymousAuthentication"); }
+            set { ResourceItemDataSource.Properties.SetItem("_rpUseAnonymousAuthentication", value); }
+        }
+
+        public void AddHeader(HeaderType headerType, string value)
+        {
+            var propertyKey = "Headers";
+
+            var headerValue = $"{AddDashesToEnumName(headerType.ToString())}={value}";
+
+            if (!ResourceItemDataSource.Properties.ContainsKey(propertyKey))
+            {
+                ResourceItemDataSource.Properties.Add(propertyKey, new List<string> { headerValue });
+            }
+            else
+            {
+                var headers = (List<string>)ResourceItemDataSource.Properties[propertyKey];
+                headers.Add(headerValue);
+            }
+        }
+
+        public void UseCsv()
+        {
+            ClearJsonConfig();
+            DataSource = new CsvDataSource();
+            ResourceItemDataSource.Properties.SetItem("Result-Type", ".csv");
+        }
+
+        public void UseExcel(string sheet = null, ExcelFileType fileType = ExcelFileType.Xlsx)
+        {
+            ClearJsonConfig();
+            DataSource = new ExcelDataSource();
+
+            var fileExt = fileType == ExcelFileType.Xlsx ? ".xlsx" : ".xls";
+            ResourceItemDataSource.Properties.SetItem("Result-Type", fileExt);
+
+            if (sheet != null)
+                Properties.SetItem("Sheet", sheet);
+        }
+
+        protected void InitializeResourceItem(string title)
+        {
+            ResourceItemDataSource = new DataSource { Provider = DataSourceProvider.WebService };
+            ResourceItem = new DataSourceItem
+            {
+                DataSource = ResourceItemDataSource,
+                DataSourceId = ResourceItemDataSource.Id,
+                Title = title
+            };
+
+            ResourceItemDataSource = ResourceItemDataSource;
+            ResourceItem = ResourceItem;
+        }
+
+        protected override void OnFieldsPropertyChanged(List<IField> fields)
+        {
+            Parameters.Add("config", BuildConfig(fields));
+        }
+
+        private Dictionary<string, object> BuildConfig(IEnumerable<IField> fields)
```

Comment:
I am not sure but when I don’t configure the fields config in the Parameters property of the DatasourceItem, 
it will cause an error, preventing the visualization from being loaded in the JSON format